corehq/apps/linked_domain/views.py

```diff
@@ -446,6 +440,27 @@ def create_remote_report_link(self, in_data):
             return {'success': False}
 
 
+def handle_create_domain_link_request(couch_user, upstream_domain, downstream_domain):
```

Comment:
Could this just be called `create_domain_link`?

Reply:
...now I see that this is called from within a `create_domain_link` function.

I'd probably have put all of this logic in `link_domains` as exceptions, which would eliminate this function. It's partly just a difference in style - I'd rather work with exception rather than functions that return error strings. In this case, `link_domains` already does some exception-based validation. If you keep this as its own function,  I don't have an obviously better name. It could be `check_create_domain_link` and not do the actual link creation, which you're patching in the tests anyway. Or feel free to leave it as is.

Reply:
I do like the idea of just returning the exception, and otherwise returning the actual link (that way I don't have to fetch it which I found to be annoying). The "downside" is there won't be test coverage for content of error messages, but that seems like a fine thing for our product.